site/en/docs/extensions/reference/action/index.md

```diff
@@ -2,9 +2,165 @@
+context.clearRect(0, 0, 16, 16);
+context.fillStyle = '#00FF00';  // Green
+context.fillRect(0, 0, 16, 16);
+const imageData = context.getImageData(0, 0, 16, 16);
+chrome.action.setIcon({imageData: imageData}, () => { ... });
+```
+
+!!!.aside
+The `action.setIcon()` API is intended to set a static image. It should not
+be used to simulate animation.
+!!!
+
+#### Formats
+
+For packed extensions (installed from a .crx file), images can be in most
+formats that the Blink rendering engine can display, including PNG, JPEG,
+BMP, ICO, and others (SVG is not supported). Unpacked extensions must use
+images in the PNG format.
+
+### Tooltip (title)
+
+The tooltip, or title, is displayed when the user hovers the mouse of the
+extension's icon in the toolbar. It is also included in the accessible text
+spoken by screenreaders when the button is focused.
+
+The default tooltip is set from the `default_title` field in the `action`
+`manifest.json` file. It can also be set programmatically with the
+`action.setTitle()` method.
+
+### Badge
+
+Actions can optionally display a "badge" &mdash; a bit of text layered over the
+icon. This makes it easy to update the action to display a small amount of
+information about the state of the extension, such as a counter. The badge has a
+text component and a background color.
+
+Note that the badge has limited space, and should typically use four characters
+or fewer.
+
+The badge does not have a default taken from the manifest; you can set it
+programmatically with `action.setBadgeBackgroundColor()` and
+`action.setBadgeText()`. When setting the color, the values can be either
+an array of four integers in the range of 0 - 255 that make up the RGBA
+color of the badge or a string with a CSS hex color value.
+
+```js
+chrome.action.setBadgeBackgroundColor(
+    {color: [0, 255, 0, 0]},  // Green
+    () => { ... });
+
+chrome.action.setBadgeBackgroundColor(
+    {color: '#00FF00'}  // Also green
+    () => { ... });
+```
+
+### Popup
+
+An action's popup will be shown when the user clicks on the extension's button
+in the toolbar. The popup can contain any HTML contents you like, and will be
+sized to fit its contents (though it has a maximum size).
+
+The popup is initially set from the `default_popup` property in the `action`
+key in the `manifest.json` file. If present, this should point to a relative
+path within the extension directory. It can also be updated dynamically to
+point to a different relative path using the `action.setPopup()` method.
+
+!!!.aside
+The `action.onClicked` event will not be dispatched if the extension action
+has a popup on the current tab.
 !!!
+
+## Per-tab state
+
+Extension actions can have different states for each tab. For instance, you
+could set the badge text to be different on each tab (to show tab-specific state).
+You can set the value for an individual tab using the `tabId` property in the
+different setting methods on the `action` API. For instance, to set the badge
```

Comment:
Hmmm. "different setting methods" reads wrong. Consider one of:

"...the various setting methods on..."
"...each of the setting methods on..."

Reply:
I like "the various setting".  Done